File: ChattingApplicationProject/Models/Message.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ChattingApplicationProject.Models
{
    public class Message
    {
        public int Id { get; set; }
        public int SenderId { get; set; }
        public string SenderUsername { get; set; }
        public AppUser Sender { get; set; }
        public int RecipientId { get; set; }
        public string RecipientUsername { get; set; }
        public AppUser Recipient { get; set; }
        public string Content { get; set; }
        public string? Emoji { get; set; }
        public string? VoiceUrl { get; set; }
        public int? VoiceDuration { get; set; } // Duration in seconds
        public string MessageType { get; set; } = "text"; // "text", "voice", "emoji"
        public DateTime? DateRead { get; set; }
        public DateTime MessageSent { get; set; } = DateTime.UtcNow;
        public bool SenderDeleted { get; set; }
        public bool RecipientDeleted { get; set; }
    }
}

File: AngularChattingAppFrontEnd/src/_models/message.ts
export interface MessageDto {
  id: number;
  senderId: number;
  senderUsername: string;
  senderPhotoUrl?: string;
  recipientId: number;
  recipientUsername: string;
  recipientPhotoUrl?: string;
  content: string;
  emoji?: string;
  voiceUrl?: string;
  voiceDuration?: number;
  messageType?: string;
  messageSent: Date;
  dateRead?: Date;
}
